eventuals/task.h

```diff
@@ -238,34 +254,56 @@ struct _TaskFromToWith {
     K_ k_;
     std::tuple<Args_...> args_;
 
-    Callback<
-        Action,
-        std::optional<std::exception_ptr>&&,
-        Args_&&...,
-        std::optional<
-            std::conditional_t<
-                std::is_void_v<From_>,
-                std::monostate,
-                From_>>&&,
-        std::unique_ptr<void, Callback<void*>>&,
-        Interrupt&,
+    // The 'dispatch_' is a `std::variant` because its either a function that
+    // creates an eventual or the value from 'Task::Success' that should be
+    // passed on to the continuation.
+    std::variant<
```

Comment:
Gosh, this is a long declaration line.

Some local (private) typedefs might help make this much more readable, e.g., one typedef for the Callback (not sure if any others make sense to pull out)